### ISSUE TITLE
Validation fails

### ISSUE THREAD

**Issue by ghost** [aws/amazon-cognito-identity-js#567]
The following attributes are required by our user pool:  `email`, `family_name` and `given_name`. 

The docs say that
> When a standard attribute is marked as required, a user cannot register unless a value for the attribute is provided. Administrators can create users without giving values for required attributes by using the AdminCreateUser API. An attribute cannot be switched between required and not required after a user pool has been created.

I use the `signUp` method to register the user, and it passes the validation only with `email` and `password`. Does `signUp` use the `AdminCreateUser` API? Shouldn't it be a simple user registration? 

```
this.userPool.signUp(email, password, attributeList, null, (error, result) => {
        if (error) {
          reject(error);
          return;
        }
        resolve(result);
});
```

**Comment by yuntuowang** [comment]
Yes, signUp doesn't use the AdminCreateUser API. The expected use case is that it will fail during validation. Can you try again to see if there is still any issue? If there is, can you provide us what is the exception, request id and the timestamp.
